quodlibet/library/libraries.py

```diff
@@ -823,6 +824,74 @@ def remove_masked(self, mount_point):
 
         self._masked.pop(mount_point, {})
 
+    def move_root(self, old_root: str, new_root: fsnative) \
+            -> Generator[None, None, None]:
+        """
+        Move the root for all songs in a given (scan) directory.
+
+        We avoid dereferencing the destination, to allow users things like:
+          1. Symlink new_path -> old_root
+          2. Move QL root to new_path
+          3. Remove symlink
+          4. Move audio files: old_root -> new_path
+
+        """
+        old_path = Path(normalize_path(old_root, canonicalise=True)).expanduser()
+        new_path = Path(normalize_path(new_root)).expanduser()
+        if not old_path.is_dir():
+            raise ValueError(f"Source {old_path!r} is not a directory")
+        if not new_path.is_dir():
+            raise ValueError(f"Destination {new_path!r} is not a directory")
+        print_d(f"{self._name}: checking {len(self.values())} item(s) for {old_path!r}")
+        missing: Set[AudioFile] = set()
+        changed = set()
+        total = len(self)
+        if not total:
+            return
+        with Task(_("Library"), _("Moving library files")) as task:
+            yield
+            for i, song in enumerate(list(self.values())):
+                task.update(i / total)
+                key = normalize_path(song.key)
+                path = Path(key)
+                if old_path in path.parents:
+                    # TODO: more Pathlib-friendly dir replacement...
+                    new_key = normalize_path(key.replace(str(old_path),
+                                                         str(new_path),
+                                                         1),
```

Comment:
I think the `key.replace` part would look better on a new line, but that's just my preference for limiting multi line statements.

Reply:
:+1: